src/map/packet_system.cpp

```diff
@@ -3487,8 +3487,15 @@ void SmallPacket0x050(map_session_data_t* const PSession, CCharEntity* const PCh
     }
 
     charutils::EquipItem(PChar, slotID, equipSlotID, containerID); // current
-    charutils::SaveCharEquip(PChar);
-    charutils::SaveCharLook(PChar);
+    // Only save char look to db every X seconds (or more) to avoid db spam when changing individual pieces
+    // Do not save equipment at all since SaveCharEquip is a very expensive operation
+    auto lastSwapTime   = PChar->GetLocalVar("core-LastGearSwap");
+    auto timeNowSeconds = std::chrono::time_point_cast<std::chrono::seconds>(server_clock::now());
+    if (lastSwapTime == 0 || (timeNowSeconds.time_since_epoch().count() - lastSwapTime) > 30)
```

Comment:
Free optimization: lockstyled players will never need look updates unless their main/sub ranged/ammo change to different skill types

Reply:
Yes looks like savecharlook doesn't return if lockstyled, so we could add a check here. Though i'm not sure how weapons fall into the equation. savecharlook is very cheap, and the way I have this is that it only runs savecharlook if it's been 30s since your last gearswap (if you gearswap every 10s doing random things the call won't happen)